automation_tools/__init__.py

```diff
@@ -370,6 +370,45 @@ def setup_fake_manifest_certificate(certificate_url=None):
     manage_daemon('restart', 'tomcat6' if distro_info()[1] <= 6 else 'tomcat')
 
 
+def setup_firewall():
+    """Setup firewall rules that Satellite 6 needs to work properly"""
+    ports = (
+        # Port 443 for HTTPS (secure WWW) must be open for incoming
+        # connections.
+        443,
+        # Port 5671 must be open for SSL communication with managed systems.
+        5671,
+        # Port 80 for HTTP (WWW) must be open to download the bootstrap files.
+        80,
+        # Port 8140 must be open for incoming Puppet connections with the
+        # managed systems.
+        8140,
+        # Port 9090 must be open for Foreman Smart Proxy connections with the
+        # managed systems.
+        9090,
+        # Port 22 must be open for connections via ssh
+        22,
+        # Port 5000 must be open for Docker registry communication.
+        5000,
+    )
+
+    for port in ports:
+        rule_exists = run(
+            r'iptables -nL | grep -E "^ACCEPT\s+tcp.*{0}"'.format(port),
```

Comment:
I'm using -E with allows me to use the extended regex support :smile: 

Will change to look for the INPUT chain. And the regex does not end with $ because rules added by the puppet module is like this `ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0           multiport ports 80,443 /* 100 accept http/https */`

About `--check` iptables option will try to improve using that later.